release-23.2: release: use license from its new location

jlinder opened the pull request:
Backport:
  * 1/1 commits from "release: use license from its new location" (#132051)
  * 1/1 commits from "release: fix location of THIRD-PARTY-NOTICES.txt" (#132175)

Please see individual PRs for details.

/cc @cockroachdb/release

---

The release process includes the LICENSE file in the tarballs and docker images that get built. This commit now uses the LICENSE file from its new location (in the repository base directory).

This commit also updates the release process to explicitly reference the THIRD-PARTY-NOTICES.txt file throughout the build process (vs. copying the licenses directory in some places before pulling the THIRD-PARTY-NOTICES.txt later in the process).

Part of RE-658

Release note: none

---

Release justification: The release process needs to use the license from the new location.

jlinder committed:
release: use license from its new location

The release process includes the LICENSE file in the tarballs and docker
images that get built. This commit now uses the LICENSE file from its
new location (in the repository base directory).

This commit also updates the release process to explicitly reference the
THIRD-PARTY-NOTICES.txt file throughout the build process (vs. copying
the licenses directory in some places before pulling the
THIRD-PARTY-NOTICES.txt later in the process).

Part of RE-658

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132145)
<!-- Reviewable:end -->

jlinder committed:
release: fix location of THIRD-PARTY-NOTICES.txt

The wrong location for THIRD-PARTY-NOTICES.txt was used in two release
scripts. Here we fix this to point at the correct location.

Part of RE-658

Release note: none

rail committed:
ci: fix location of licenses for docker image

Part of RE-658
Release note: none
